### PR TITLE
[ENG-3724] Delete multiple files

### PR DESCRIPTION
**Pull request by adlius**
-   Ticket: []
-   Feature flag: n/a

### CODE REVIEW

File: lib/osf-components/addon/components/file-browser/delete-file-modal/component.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import { taskFor } from 'ember-concurrency-ts';
+import File from 'ember-osf-web/packages/files/file';
+import captureException from 'ember-osf-web/utils/capture-exception';
+
+interface Args {
+    items: File[];
+    reload: () => {};
+}
+
+export default class FileBrowser extends Component<Args> {
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/file-browser/delete-file-modal/component.ts:15]:
```suggestion
export default class DeleteFileModal extends Component<Args> {
```

File: lib/osf-components/addon/components/file-browser/delete-file-modal/template.hbs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                        <div>
+                            {{#if item.isFile}}
+                                <FaIcon @icon='file-alt' />
+                            {{else}}
+                                <FaIcon @icon='folder' />
+                            {{/if}}
+                            <span local-class='ItemName'>{{item.name}}</span>
+                        </div>
+                        <Button
+                            @layout='fake-link'
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/file-browser/delete-file-modal/template.hbs:34]: This guy needs an aria-label

File: lib/osf-components/addon/components/file-browser/template.hbs [hunk 3]
```diff
@@ -111,14 +111,7 @@
...
                 </Button>
-                <Button
-                    @layout='fake-link'
-                >
-                    <FaIcon
-                        @icon='trash'
-                    />
-                    {{t 'general.delete'}}
-                </Button>
+                <FileBrowser::DeleteFileModal @items={{@manager.selectedFiles}} @reload={{@manager.reload}} />
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/file-browser/template.hbs:114]: There should be a permission check here

File: lib/osf-components/addon/components/file-browser/delete-file-modal/template.hbs [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+                    this.reset
+                    dialog.close
+                )}}
+            >
+                {{t 'general.done'}}
+            </Button>
+        {{else}}
+            <Button
+            @type='secondary'
+            {{on 'click' dialog.close}}
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/file-browser/delete-file-modal/template.hbs:86]: When a user hits cancel, should we stop the delete tasks that are currently in progress? Also needs one more indent level

**Review by futa-ikeda — CHANGES_REQUESTED** [review]

File: lib/osf-components/addon/components/file-browser/delete-file-modal/template.hbs [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                                <FaIcon @icon='file-alt' />
+                            {{else}}
+                                <FaIcon @icon='folder' />
+                            {{/if}}
+                            <span local-class='ItemName'>{{item.name}}</span>
+                        </div>
+                        <Button
+                            @layout='fake-link'
+                            {{on 'click' (perform this.retryDelete item)}}
+                        >
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/file-browser/delete-file-modal/template.hbs:36]: This should not be an active button if the retryDelete task is running. It should probably be a pulsed spinner with no clickability.

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
Just one thing from me.

**Review by brianjgeiger — APPROVED** [review]
My stuff is cleared up. This is good once Futa is happy.

**Review by futa-ikeda — APPROVED** [review]

**Pull request merged**